torch/fx/passes/backends/cudagraph_partition.py

```diff
@@ -0,0 +1,232 @@
+import dataclasses
+from contextlib import ExitStack
+from typing import Any, Callable, Optional
+from unittest.mock import patch
+
+import torch
+from torch import fx
+from torch._library.utils import lookup_op
+
+
+# Part1: Split gm
+def resolve_defined_ops(op_names: list[str]) -> list["torch._ops.OpOverload"]:
+    """Resolve operator names to OpOverload objects.
+
+    Skips operators that fail to resolve (e.g., operators not registered or
+    model-specific operators not present in the current model).
+
+    Note: Users should inspect the operator graph before lowering and ensure
+    the specified operators are present in the final graph. Built-in PyTorch
+    operators (aten::*, torch::*) may be decomposed, fused, or transformed
+    during Inductor's compilation passes, so use them with caution.
+
+    Args:
+        op_names: List of operator names in PyTorch format
+            (e.g., "namespace::op_name")
+
+    Returns:
+        List of successfully resolved operator overloads
+    """
+    resolved = []
+    for op_name in op_names:
+        try:
+            resolved.append(lookup_op(op_name))
+        except Exception:
+            # Skip operators that don't exist (e.g., model-specific ops)
+            continue
+
+    return resolved
+
+
+@dataclasses.dataclass
+class SplitItem:
+    submod_name: str
+    graph_id: int
+    is_cudagraph_unsafe: bool
+    gm: fx.GraphModule
+
+
+def split_graph(
+    gm: fx.GraphModule, resolved_ops: list[torch._ops.OpOverload]
+) -> tuple[fx.GraphModule, list[SplitItem]]:
+    """
+    Given a graph module and a list of ops, split the gm into subgraphs.
+
+    @param gm: the graph module to be split
+    @param resolved_ops: a list of ops to be split
+
+    @return: a tuple of (split_gm, split_items)
+    """
+    # split graph by ops
+    subgraph_id = 0
+    node_to_subgraph_id = {}
+    split_op_graphs = []
+    for node in gm.graph.nodes:
+        if node.op in ("output", "placeholder"):
+            continue
+        # Match node.target against resolved_ops
+        # node.target can be OpOverloadPacket, need to check .default
+        if node.op == "call_function" and (
+            node.target in resolved_ops
+            or (hasattr(node.target, "default") and node.target.default in resolved_ops)
+        ):
+            subgraph_id += 1
```

Comment:
if you have a few resolved_ops in a sequence, subgraph_id will keep increasing. 
but it's not indicating a new cuda graph region